ivy/func_wrapper.py

```diff
@@ -962,34 +962,44 @@ def _wrap_function(
         # set attributes
         for attr in original.__dict__.keys():
             # private attribute or decorator
-            if attr.startswith("_") or hasattr(ivy, attr) or attr == "handles_out_arg":
+            if (
+                attr.startswith("_")
+                or hasattr(ivy, attr)
+                or (not compositional and attr == "mixed_function")
+            ):
                 continue
             setattr(to_wrap, attr, getattr(original, attr))
         # Copy docstring
         docstring_attr = ["__annotations__", "__doc__"]
         for attr in docstring_attr:
             setattr(to_wrap, attr, getattr(original, attr))
-        # wrap decorators
-        mixed = hasattr(original, "mixed_function")
-        if mixed:
-            to_replace = {
-                True: ["inputs_to_ivy_arrays"],
-                False: [
-                    "outputs_to_ivy_arrays",
-                    "inputs_to_native_arrays",
-                ],
-            }
-            # if the backend has a primary implementation
-            # we'll store the compositional fn's reference
-            # for the handle_mixed_function decorator
-            if to_wrap != original:
-                to_wrap.compos = original
-            for attr in to_replace[compositional]:
-                setattr(original, attr, True)
+        # to_native_arrays_and_back must be added
+        # to the primary implementation (if it exists)
+        # of mixed functions.
+        if original != to_wrap:
+            if not (
```

Comment:
I think we could simply check if `inputs_to_ivy_arrays` has been applied rather than checking if the other 2 aren't applied, as every compositional implementation should have the `inputs_to_ivy_arrays` decorator

Reply:
Oh, you're right! If primary implementation exists ( i.e `to_wrap != original`)  and `original` has an `input_to_ivy_arrays` attribute then that means it's a mixed function. Have updated it now.